src/TodoContext.tsx

```diff
@@ -0,0 +1,90 @@
+import { createContext, useContext, useState } from 'react';
+import { CompletionStatus } from './types/CompletionStatus';
+import { Todo } from './types/Todo';
+import { countItemsLeft } from './utils/countItemsLeft';
+
+type TodoProviderProps = {
+  children: React.ReactNode;
+};
+
+type TodosContext = {
+  todos: Todo[];
+  titleField: string;
+  errorMessage: string;
+  tempTodo: Todo | null;
+  filterByStatus: CompletionStatus;
+  loadingItemsIds: number[];
+  itemsLeft: number;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setTitleField: React.Dispatch<React.SetStateAction<string>>;
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  setFilterByStatus: React.Dispatch<React.SetStateAction<CompletionStatus>>;
+  setLoadingItemsIds: React.Dispatch<React.SetStateAction<number[]>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  handleError: (errMessage: string) => void;
+};
+
+const initialState = {
+  todos: [],
+  titleField: '',
+  errorMessage: '',
+  tempTodo: null,
+  filterByStatus: CompletionStatus.All,
+  loadingItemsIds: [],
+  itemsLeft: 0,
+  setTodos: () => {},
+  setTitleField: () => {},
+  setTempTodo: () => {},
+  setFilterByStatus: () => {},
+  setErrorMessage: () => {},
+  handleError: () => {},
+  setLoadingItemsIds: () => {},
+};
+
+export const TodoContenxt = createContext<TodosContext>(initialState);
```

Comment:
What's the difference between `TodoContext` and `TodosContext` on 10 line?

```suggestion
export const TodoContext = createContext<TodosContext>(initialState);
```